com.unity.shadergraph/Editor/GraphUI/EditorCommon/Preview/PreviewManager.cs

```diff
@@ -1,40 +1,45 @@
 ﻿using System;
 using System.Collections.Generic;
-using Editor.GraphUI.Utilities;
+using UnityEditor.GraphToolsFoundation.Overdrive;
 using UnityEditor.ShaderGraph.GraphDelta;
-using UnityEditor.ShaderGraph.GraphUI.DataModel;
 using UnityEngine.GraphToolsFoundation.Overdrive;
 using UnityEngine;
-using UnityEngine.Rendering;
 
-namespace UnityEditor.ShaderGraph.GraphUI.EditorCommon.Preview
+namespace UnityEditor.ShaderGraph.GraphUI
 {
     using PreviewRenderMode = HeadlessPreviewManager.PreviewRenderMode;
 
     public class PreviewManager
     {
         HeadlessPreviewManager m_PreviewHandlerInstance;
 
+        GraphViewStateComponent m_GraphViewStateComponent;
+
         HashSet<string> m_DirtyNodes;
 
         ShaderGraphModel m_GraphModel;
 
         Dictionary<string, SerializableGUID> m_NodeLookupTable;
 
-        public PreviewManager(ShaderGraphModel graphModel)
+        public PreviewManager(GraphViewStateComponent graphViewStateComponent)
         {
-            m_GraphModel = graphModel;
+            m_GraphViewStateComponent = graphViewStateComponent;
 
             m_PreviewHandlerInstance = new HeadlessPreviewManager();
 
-            m_PreviewHandlerInstance.SetActiveGraph(m_GraphModel.GraphHandler);
-            m_PreviewHandlerInstance.SetActiveRegistry(m_GraphModel.RegistryInstance);
-
             m_DirtyNodes = new HashSet<string>();
             m_NodeLookupTable = new Dictionary<string, SerializableGUID>();
+        }
+
+        public void Initialize(ShaderGraphModel graphModel)
+        {
+            m_GraphModel = graphModel;
```

Comment:
On initially opening a blank project and attempting to open the editor through the Window->Shaders->ShaderGraph menu, the provided graphModel is null, leading to a null exception in the following lines.

Reply:
This has been resolved with an additional null check. 

In theory the null check shouldn't be needed and we can probably remove it once we understand GTF initialization better.

Reply:
The Window > Shaders > ShaderGraph creation path is working for me now.